ci-operator/step-registry/cucushift/upgrade/tolatest/cucushift-upgrade-tolatest-commands.sh

```diff
@@ -0,0 +1,681 @@
+    fi
+    if ! oc adm release extract --to "${tobeCredsDir}" --included --credentials-requests "${TARGET}"; then
+        echo "Failed to extract CRs from tobe upgrade release payload!" && exit 1
+    fi
+
+    # TODO: add gcp and azure
+    # Update iam role with ccoctl based on tobeCredsDir
+    tmp_ret=0
+    diff -r "${preCredsDir}" "${tobeCredsDir}" || tmp_ret=1
+    if [[ ${tmp_ret} != 0 ]]; then
+        toManifests="/tmp/to-manifests"
+        mkdir "${toManifests}"
+        case "${platform}" in
+        "AWS")
+            if [[ ! -e ${SHARED_DIR}/aws_oidc_provider_arn ]]; then
+                echo "No aws_oidc_provider_arn file in SHARED_DIR" && exit 1
+            else
+                export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+                infra_name=${NAMESPACE}-${UNIQUE_HASH}
+                oidc_provider=$(head -n1 ${SHARED_DIR}/aws_oidc_provider_arn)
+                extract_ccoctl
+                if ! ccoctl aws create-iam-roles --name="${infra_name}" --region="${LEASED_RESOURCE}" --credentials-requests-dir="${tobeCredsDir}" --identity-provider-arn="${oidc_provider}" --output-dir="${toManifests}"; then
+                    echo "Failed to update iam role!" && exit 1
+                fi
+                if [[ "$(ls -A ${toManifests}/manifests)" ]]; then
+                    echo "Apply the new credential secrets."
+                    oc apply -f "${toManifests}/manifests"
+                fi
+            fi
+            ;;
+        *)
+           echo "to be supported platform: ${platform}" 
+           ;;
+        esac
+    fi
+}
+
+# Add cloudcredential.openshift.io/upgradeable-to: <version_number> to cloudcredential cluster when cco mode is manual or the case in OCPQE-19413
+function cco_annotation(){
+    if (( SOURCE_MINOR_VERSION == TARGET_MINOR_VERSION )) || (( SOURCE_MINOR_VERSION < 8 )); then
+        echo "CCO annotation change is not required in either z-stream upgrade or 4.7 and earlier" && return
+    fi
+
+    local cco_mode; cco_mode="$(oc get cloudcredential cluster -o jsonpath='{.spec.credentialsMode}')"
+    local platform; platform="$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}')"
+    if [[ ${cco_mode} == "Manual" ]]; then
+        echo "CCO annotation change is required in Manual mode"
+    elif [[ -z "${cco_mode}" || ${cco_mode} == "Mint" ]]; then
+        if [[ "${SOURCE_MINOR_VERSION}" == "14" && ${platform} == "GCP" ]] ; then
+            echo "CCO annotation change is required in default or Mint mode on 4.14 GCP cluster"
+        else
+            echo "CCO annotation change is not required in default or Mint mode on 4.${SOURCE_MINOR_VERSION} ${platform} cluster"
+            return 0
+        fi
+    else
+        echo "CCO annotation change is not required in ${cco_mode} mode"
+        return 0
+    fi
+
+    echo "Require CCO annotation change"
+    local wait_time_loop_var=0; to_version="$(echo "${DUMMY_TARGET_VERSION}" | cut -f1 -d-)"
```

Comment:
Once we can determinate the exact target version just like https://github.com/openshift/release/pull/49743/files#r1526174170, we can use `TARGET_VERSION` instead of `DUMMY_TARGET_VERSION`